resources/stats.py

```diff
@@ -0,0 +1,53 @@
+"""All the Statistics API endpoints"""
+
+from flask import current_app, make_response, jsonify
+from flask_restful import Resource
+from database.models import Quote, Author
+
+
+def get_query_most_popular(field, top_n):
+    """Returns Query to get top_n items with given field"""
+
+    return [
+        {"$project": {field: 1}},
+        {"$unwind": f"${field}"},
+        {"$group": {"_id": f"${field}", "count": {"$sum": 1}}},
+        {"$sort": {"count": -1}},
+        {"$limit": top_n},
+    ]
+
+
+class StatsApi(Resource):
+    """All the Statistics API endpoints"""
+
+    def get(self):
+        """Get Statistics for Quotes, Authors and Tags"""
+
+        try:
+            top_tags_query = get_query_most_popular("tags", 5)
+            tag_cursor = Quote.objects(author__exists=True).aggregate(top_tags_query)
+            current_app.logger.info("GET Stats - FETCHED TOP Tags")
+
+            top_author_query = get_query_most_popular("author", 1)
+            author_cursor = Quote.objects(author__exists=True).aggregate(
+                top_author_query
+            )
+            current_app.logger.info("GET Stats - FETCHED TOP Author")
+
+            top_tags = [tag["_id"] for tag in tag_cursor]
+            top_author = str(list(author_cursor)[0]["_id"])
```

Comment:
I am assuming that this gets the IDs and not the actual data. If its true then try getting list of tag names and author name.

Reply:
Tag is actually getting the tag name but author is getting the author ID.
because in quotes collection we have author id we will need to do one more fetch from author collection. that's why I just used ID for author.

Reply:
should I get only name of author.

Reply:
Check below comment for the requirements

Reply:
ok. but along with that should I also include name or id is enough?

Reply:
Include name, no id should be returned. Only data.